src/test/java/com/google/sps/TextAnalyserTest.java

```diff
@@ -0,0 +1,132 @@
+    } catch (IOException e) {
+      fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void analysePositiveMood() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("Happy Birthday!");
+    String expected = "thrilled";
+    String actual = textAnalyser.getMood();
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void analyseNegativeMood() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("Smoking, well, there is no healthy amount of smoking." +
+                                                  "Healthy here meant never having smoked.");
+    String expected = "depressed";
+    String actual = textAnalyser.getMood();
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testForCorrectEvents() {
+    TextAnalyser textAnalyser = new TextAnalyser("Today is my Birthday. Today you got a job promotion.");
+    Set<String> expected = new HashSet<>();
+    Set<String> actual = textAnalyser.getEvents();
+
+    expected.add("birthday");
+    expected.add("job");
+    expected.add("promotion");
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testEntityThatReturnsLocation() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("I am going to Paris this weekend.");
+    assertTrue("There is no city displayed.", textAnalyser.getEntities().contains("paris"));
+  }
+
+  @Test
+  public void testEntityThatReturnsName() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("Hello, Mark!");
+    assertTrue("There is no name displayed", textAnalyser.getEntities().contains("mark"));
+  }
+
+  @Test
+  public void testEntityThatReturnsEvent() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("My wedding is coming soon!");
+    assertTrue("There is no event displayed.", textAnalyser.getEntities().contains("wedding"));
+  }
+
+  @Test
+  public void testOnlyLowerCaseKeyWords() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("Hello, Mark! My Birthday is coming very soon and I want to" +
+                                                  " invite you to my Party! It is in Paris. It is not expensive." +
+                                                  " We will have so much fun! All of our friends will also come to the party");
+    Set<String> expected = new HashSet<>();
+    Set<String> actual = textAnalyser.getKeyWords();
+
+    for(String keyWord : actual) {
+      expected.add(keyWord.toLowerCase());
+    }
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testCorrectKeyWords() throws IOException {
+    TextAnalyser textAnalyser = new TextAnalyser("Happy Birthday! We will go to Paris!");
+    Set<String> expected = new HashSet<>();
+    Set<String> actual = textAnalyser.getKeyWords();
+
+    expected.add("birthday");
+    expected.add("paris");
+    expected.add("serene");
+
+    assertEquals(expected, actual);
```

Comment:
I find `assertTrue(keywords.isEmpty())` easier to read.

(out of scope, but good to know: there is Hamcrest which allows to express assertions in more expressive language, like `assertThat(keywords, is(empty()));` which furthermore provides much more expressive error messages (rather than "assertion is false").